### PR TITLE
Add GH action for GitLeaks

### PR DESCRIPTION
**Pull request by yotommy**
### Description

Add [GitHub Action](https://github.com/gitleaks/gitleaks-action) for [GitLeaks](https://github.com/gitleaks/gitleaks).

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [ ] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### Issue(s) this completes

Relates to #7099

### CODE REVIEW

**Comment by yotommy** [comment]
* **#7480** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7480?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yotommy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/gitleaks.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: gitleaks
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *" # run once a day at 4 AM
```
Comment (gwendolyngoetz) [inline .github/workflows/gitleaks.yaml:7]: (blocking) GitHub actions run in UTC. Can you update the time to be 4 AM EST/EDT and update the comment to include the time it runs in EST/EDT and PST/PDT. It make it less confusing that having to try to remember what it should be running at each time you open the file.

File: .github/workflows/gitleaks.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  scan:
+    name: gitleaks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: gitleaks/gitleaks-action@v2.3.4
```
Comment (gwendolyngoetz) [inline .github/workflows/gitleaks.yaml:16]:
```suggestion
      - uses: gitleaks/gitleaks-action@v2
```
(blocking) Typically github actions are just tagged at the major verson

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: .github/workflows/gitleaks.yaml
Comment (gwendolyngoetz) [inline .github/workflows/gitleaks.yaml:1]: (blocking) Like the name in the yaml, please change this to `scan-for-secrets.yaml`

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: .github/workflows/gitleaks.yaml (hunk 1 above)
Comment (yotommy) [reply .github/workflows/gitleaks.yaml]: Done

File: .github/workflows/gitleaks.yaml
Comment (yotommy) [reply .github/workflows/gitleaks.yaml:1]: Done

File: .github/workflows/gitleaks.yaml (hunk 1 above)
Comment (yotommy) [reply .github/workflows/gitleaks.yaml]: Done

**Review by gwendolyngoetz — DISMISSED** [review]

File: .github/workflows/scan_for_secrets.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Scan for Secrets
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 9 * * *" # run daily at 4AM EST/5AM EDT/1AM PST/2AM PDT
```
Comment (dkatzz) [inline .github/workflows/scan_for_secrets.yaml:7]: **nitpick:** you could consider putting the comment above so it fits in one line. Also extremely nitpicky but maybe add a space before and after the "/"s

**Review by dkatzz — DISMISSED** [review]

File: .github/workflows/scan_for_secrets.yaml (hunk 3 above)
Comment (yotommy) [reply .github/workflows/scan_for_secrets.yaml]: Done

**Pull request merged**